lib/fs.js

```diff
@@ -268,6 +269,26 @@ function handleError(val, callback) {
   return false;
 }
 
+function openWrap(path, flags, mode, req) {
+  var longPath = pathModule._makeLong(path);
+  if (isWindows) {
+    // \\?\c:\somepath\1:path
+    // \\?\UNC\somepath\2:path
+    if (longPath.indexOf(':', 6) !== -1) {
+      const err = new errors.Error('ERR_INVALID_WIN32_PATH', longPath);
+      if (req) {
+        return process.nextTick(function() {
```

Comment:
I'm not sure this extra closure is needed. Perhaps it could just be `process.nextTick(req.oncomplete, err)`.